docs/API/V1/server_config.md

```diff
@@ -0,0 +1,253 @@
+a `message` field.
+
+Any additional JSON data may be provided, and will be returned uninterpreted
+when a client requests it with `GET /api/v1/server/configuration/server-banner`.
+
+For example,
+
+```
+{
+    "server-banner": {
+        "message": "The server will be down for 2 hours on Monday, July 41st",
+        "contact": {
+            "email": "admin@pbench.example.com",
+            "phone": "(555) 555-5555",
+            "hours": ["8:00 EST", "17:00 EST"]
+        },
+        "statistics": {
+            "datasets": 50000,
+            "hours-up": 2.3,
+            "users": 26.4
+        }
+    }
+}
+```
+
+### server-state
+
+This server configuration setting allows a server administrator to control
+the operating state of the server remotely. As for
+[`server-banner`](#server-banner), the value is a JSON object, and any JSON
+fields passed in to the server will be returned to a client, however the
+following fields have special meaning:
+
+**`status`** \
+The operating status of the server.
+
+* `enabled`: Normal operation.
+* `disabled`: Most server API endpoints will fail with the **503** (service
+unavailable) HTTP status. However a few endpoints are always allowed:
+  * [login](./login.md) because only an authenticated user with `ADMIN` role
+    can modify server configuration settings;
+  * [logout](./logout.md) for consistency;
+  * [endpoints](./endpoints.md) for server configuration information;
+  * [server_configuration](./server_config.md) to allow re-enabling the server
+    or modifying the banner.
+* `readonly`: The server will respond to `GET` requests for information, but will return **503** (service unavailable) for any attempt to modify server state. (With the same exceptions as listed above for `disabled`.)
+
+**`message`** \
+A message to explain downtime. This is required when the `status` is `disabled`
+or `readonly` and optional otherwise. Note that a `message` when the `status`
+is `enabled` won't normally be seen, unless a client asks for the `server-state`
+configuration setting. The `server-banner` configuration setting is generally
+more appropriate under normal operating conditions.
+
+When the `status` is `disabled` or `readonly`, and an API call fails with **503**
+(service unavailable), the full JSON object will be returned as the
+`application/json` error response body. A `message` key in an error response is
+a standard convention, and many clients will display this as an explanation for
+the failure. The client also has access to any additional information provided
+in the `server-state` JSON object including the `status`.
+
+## Response body
+
+The `application/json` response body for `PUT` is exactly the same as for
+[`GET`](#response-body) when the same server configuration settings are
+requested, showing only the server configuration settings that were changed
+in the `PUT`.
+
+```
+PUT /api/v1/server/configuration/dataset-lifetime?value=4
+
+{
+    "dataset-lifetime": "4"
+}
```

Comment:
It is sufficiently obvious that the thing after the blank line (here and at line 249) is the _response_ to the preceding request (and not, e.g., the request _body_)?  (If so, then I'll wander away, but, if not, perhaps some sort of labeling would be good?)

Reply:
I thought it was, but if you think it's ambiguous I suppose I was wrong. I thought that trying to insert labels would be awkward and harder to read, but I'll give it a try.